marklogic-data-hub/src/main/java/com/marklogic/hub/step/MarkLogicStepDefinitionProvider.java

```diff
@@ -0,0 +1,24 @@
+package com.marklogic.hub.step;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.io.JacksonHandle;
+
+public class MarkLogicStepDefinitionProvider implements StepDefinitionProvider {
+
+    private DatabaseClient databaseClient;
+
+    public MarkLogicStepDefinitionProvider(DatabaseClient databaseClient) {
+        this.databaseClient = databaseClient;
+    }
+
+    @Override
+    public StepDefinition getStepDefinition(String name, StepDefinition.StepDefinitionType type) {
+        // TODO Support custom step definitions
```

Comment:
This won't work yet for custom step definitions, but this is just a prototype. The real impl will support them.